web/ui/react-app/src/pages/TSDBStatus.test.tsx

```diff
@@ -0,0 +1,211 @@
+import { faSpinner } from '@fortawesome/free-solid-svg-icons';
+import { TSDBStatus } from '.';
+import { TSDBMap } from './TSDBStatus';
+
+const fakeTSDBStatusResponse: {
+  status: string;
+  data: TSDBMap;
+} = {
+  status: 'success',
+  data: {
+    CardinalityLabelStats: [
+      {
+        Name: '__name__',
+        Count: 5,
+      },
+    ],
+    CardinalityMetricsStats: [
+      {
+        Name: 'scrape_duration_seconds',
+        Count: 1,
+      },
+      {
+        Name: 'scrape_samples_scraped',
+        Count: 1,
+      },
+    ],
+    LabelValueStats: [
+      {
+        Name: '__name__',
+        Count: 103,
+      },
+    ],
+    LabelValuePairsStats: [
+      {
+        Name: 'instance=localhost:9100',
+        Count: 5,
+      },
+    ],
+  },
+};
+
+describe('TSDB Stats', () => {
+  beforeEach(() => {
+    fetch.resetMocks();
+  });
+
+  it('before data is returned', () => {
+    const tsdbStatus = shallow(<TSDBStatus />);
+    const icon = tsdbStatus.find(FontAwesomeIcon);
+    expect(icon.prop('icon')).toEqual(faSpinner);
+    expect(icon.prop('spin')).toEqual(true);
+  });
+
+  describe('when data is returned', () => {
+    it('renders tables', async () => {
+      const mock = fetch.mockResponse(JSON.stringify(fakeTSDBStatusResponse));
+      let page: ReactWrapper;
+
+      await act(async () => {
+        page = mount(<TSDBStatus pathPrefix="/path/prefix" />);
+      });
+      page.update();
+
+      expect(mock).toHaveBeenCalledWith('/path/prefix/api/v1/status/tsdb', undefined);
+      const tables = page.find(Table);
+      const stats = Object.keys(fakeTSDBStatusResponse.data);
+      expect(stats.length).toBe(tables.length);
+    });
+  });
+
+  describe('when an error is returned', () => {
+    it('displays an alert', async () => {
+      const mock = fetch.mockReject(new Error('error loading flags'));
+
+      let flags: ReactWrapper;
+      await act(async () => {
+        flags = mount(<TSDBStatus pathPrefix="/path/prefix" />);
+      });
+      flags.update();
+
+      expect(mock).toHaveBeenCalledWith('/path/prefix/api/v1/status/tsdb', undefined);
+      const alert = flags.find(Alert);
+      expect(alert.prop('color')).toBe('danger');
+      expect(alert.text()).toContain('error loading flags');
+    });
+  });
+
+  describe('Table Data Validation', () => {
+    it('Highest Cardinality Labels Table', async () => {
+      const mock = fetch.mockResponse(JSON.stringify(fakeTSDBStatusResponse));
+      const data = fakeTSDBStatusResponse.data.CardinalityLabelStats;
+      let page: ReactWrapper;
+      await act(async () => {
+        page = mount(<TSDBStatus pathPrefix="/path/prefix" />);
+      });
+      page.update();
+
+      expect(mock).toHaveBeenCalledWith('/path/prefix/api/v1/status/tsdb', undefined);
+
+      const table = page
+        .find(Table)
+        .at(0)
+        .find('tbody');
+      const rows = table.find('tr');
+
+      for (let i = 0; i < data.length; i++) {
```

Comment:
I guess this is a good snapshot test candidate :)
Next few tests as well

Reply:
thank you for the suggestion. I think snapshot tests are used when you want to make sure UI does not change. In this case, I am looking for data(mock) is appearing in the right table and column, etc.